### ISSUE TITLE
:refer doesn't allow to override existing vars

### ISSUE THREAD

**Issue by mk** [borkdude/sci#527]
**version**
git dep at 4c920c0360d7d9a2595b26a5808d2d74daed3cd8.

**platform**
OS X
**problem**
<img width="620" alt="Screen Shot 2021-02-08 at 20 40 31" src="https://user-images.githubusercontent.com/1187/107272377-e6ac2700-6a4d-11eb-9f86-012149e57886.png">

Since upgrading from 0.2.1 to 4c920c0360d7d9a2595b26a5808d2d74daed3cd8 I noticed the following change: in 0.2.1 `:refer :all` would overwrite `+` from `clojure.core`, in the latest master it would not.

I've bisected and found f8f42684e928fea6eb9516212a5e9c8fc7a73426 as the commit that introduced the change in behavior.

**repro**
Evaluating
```clojure
(ns test.repro)
(def + str)
(require '[test.repro :refer [+]])
```
followed by
```
(+ 1 2)
```
Should return `"12"`, not `3`.

**Comment by borkdude** [comment]
Small repro:

```
(sci/eval-string "(defn join []) (refer 'clojure.string) (ns-name (:ns (meta #'join)))")
```

This should return `clojure.string` instead of `user`.

**Comment by borkdude** [comment]
@mk Fixed with 55b9f556ff85df0057a5008dd0fa4677e330d606
